### ISSUE TITLE
[bug] Projects not showing deploy button with newer versions of sanity

### ISSUE THREAD

**Issue by tim-soft** [ndimatteo/sanity-plugin-vercel-deploy#60]
Hey @ndimatteo  👋

Thank you for the plugin, its incredibly useful!

Unfortunately something appears to have broken recently -- im not sure if this has to do with Vercel or newer versions of sanity but as of this writing I am trying with `sanity@3.64.3` and `sanity-plugin-vercel-deploy@3.3.4`

<img width="1057" alt="Screenshot 2024-11-25 at 8 03 01 PM" src="https://github.com/user-attachments/assets/49058fa3-cddd-408c-93d9-c0eb3b16c947">

Projects show as a list with no deployment buttons

**Comment by ndimatteo** [comment]
Hey there @tim-soft it's a bit difficult to tell what the issue is based on the info you provided.

It doesn't look as if any of the required fields were filled out properly, notably the Webhook URL appears to be a placeholder (note the: `<project-key-redacted>` bit).

If all deploy fields are filled out correctly (including a token with proper access) you shouldn't have any issue with the current version of this plugin.

If you're still experiencing issues, could you provide a bit more info on the steps you are taking?

**Comment by tim-soft** [comment]
Yes, I removed the project key for the sake of the screenshot

My project worked perfectly with this plugin, then at some point between sanity upgrades, the plugin stopped working
I tried generating new keys, the sanity/plugin config has not changed

**Comment by webbeandk** [comment]
I have this issue as well.

Did you fix this?

**Comment by ndimatteo** [comment]
Thanks for sharing more info @tim-soft!

1. Vercel [versions](https://vercel.com/docs/rest-api#versioning) their API, so there shouldn't be any issues with that aspect of the plugin using the Vercel API
2. Aside from storing the deployment fields in a Sanity document, Sanity has little to do with how this plugin functions

I've tested on a few production studios (using Sanity `v3.64` and `v3.65` alongside the latest version of the plugin) and I'm not seeing any issues with deployment entries showing up and functioning properly.

I'd double-check your Vercel Token to ensure it has proper access to the project, and whether it's part of a vercel team.

**Comment by michaelburtonray** [comment]
Same, the deploy buttons disappeared for me and I haven't changed any settings, just upgraded packages.

sanity `3.65.1`
react": `18.3.1`
sanity-plugin-vercel-deploy: `10.9.1`

**Comment by ndimatteo** [comment]
@michaelburtonray you should be on version [3.3.4](https://www.npmjs.com/package/sanity-plugin-vercel-deploy/v/3.3.4) of this plugin, that is the latest (not sure where you got `.10.9.1` from).

I'm testing on the same versions of React (`18.3.1`) and Sanity (`3.65.1`) and not seeing any errors, even with the `useColorScheme` deprecated warning (which shouldn't cause any performance issues).

**Comment by tim-soft** [comment]
I also had a working project that stopped out of the blue, I don't see any errors in chrome devtools

**Comment by sellsteam13** [comment]
I had a similar issue, and updating the "sanity" package to version 3.66.1 helped me.

**Comment by johannesmutter** [comment]
Same issue here. Downgrading React helped me too.

**Comment by davidatqala** [comment]
I am still unable to get it to work, even after the downgrade to react 18.3.1. It just shows the webhook and Im not able to edit, delete or remove them now.

**Comment by wdtamagi** [comment]
Yeah, same thing here with React 19. The problem is that I'm using NextJs 15 and it requires this version.

**Comment by alaxvong** [comment]
I'm part of the group that encountered button rendering issues after upgrading our site to React 19/Next 15. I was able to resolve it by installing/upgrading [react-is](https://www.npmjs.com/package/react-is) to version 19.0.0. I’m not sure what the root problem is, but maybe someone else can help drill down further. In the meantime, this fix has restored functionality for our client.

**Comment by davidatqala** [comment]
> I'm part of the group that encountered button rendering issues after upgrading our site to React 19/Next 15. We were able to resolve it by installing/upgrading [react-is](https://www.npmjs.com/package/react-is) to version 19.0.0. I’m not sure what the root problem is, but maybe someone else can help drill down further. In the meantime, this fix has restored functionality for our client.

this worked! thank you!

**Comment by tim-soft** [comment]
@alaxvong this also works for me!

@ndimatteo it seems like `react-is` needs to be either a dep or peer dep in this project

**Comment by tardis741** [comment]
> I'm part of the group that encountered button rendering issues after upgrading our site to React 19/Next 15. We were able to resolve it by installing/upgrading [react-is](https://www.npmjs.com/package/react-is) to version 19.0.0. I’m not sure what the root problem is, but maybe someone else can help drill down further. In the meantime, this fix has restored functionality for our client.

This worked for me too. Thanks @alaxvong !

**Comment by tillschweneker** [comment]
Installing react-is was not working for me. After adding this override to my package.json, the button reappeared. It seems, that the package should update to the latest @sanity/ui.

Using react 19.1 and sanity 3.83.0

```
"overrides": {
	"sanity-plugin-vercel-deploy": {
		"react": "^19.1.0",
		"react-dom": "^19.1.0",
		"@sanity/ui": "^2"
	}
}
```

**Comment by TamirHen** [comment]
Thanks you! I had the same problem using Next 15 and React 19.
The beta version works and looks great!
Love this plugin ⭐
